server/src/test/java/com/octopus/teamcity/opentelemetry/server/LogMaskerTest.java

```diff
@@ -0,0 +1,42 @@
+package com.octopus.teamcity.opentelemetry.server;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.Test;
+
+import java.util.Locale;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class LogMaskerTest {
+
+    @Test
+    void testMaskLowerCase() {
+        int APIKEY_LENGTH = 31;
+        String generatedString = RandomStringUtils.randomAlphanumeric(APIKEY_LENGTH).toLowerCase(Locale.ROOT);
+        assertEquals(LogMasker.API_KEY_REPLACEMENT_REGEX, LogMasker.mask(generatedString));
```

Comment:
Consider just asserting for the presence or absence of the unmasked key in the final string rather than exact matches

Reply:
add an assertion to this test case `assertFalse(LogMasker.mask(generatedString).contains(generatedString));`